smart-campus-service-impl/src/main/java/hu/unideb/smartcampus/service/api/impl/UserRegistrationServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package hu.unideb.smartcampus.service.api.impl;
+
+import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import hu.unideb.smartcampus.service.api.UserRegistrationService;
+import hu.unideb.smartcampus.service.api.UserService;
+import hu.unideb.smartcampus.service.api.domain.User;
+import hu.unideb.smartcampus.service.api.exception.RegistrationFailedException;
+import hu.unideb.smartcampus.shared.enumeration.Role;
+import hu.unideb.smartcampus.shared.exception.XmppException;
+import hu.unideb.smartcampus.webservice.api.ejabberd.XmppUserService;
+
+@Service
+public class UserRegistrationServiceImpl implements UserRegistrationService {
+
+  @Autowired
+  private XmppUserService xmppUserService;
+
+  @Autowired
+  private UserService userService;
+
+  @Override
+  public void doRegister(String username) throws RegistrationFailedException {
+    final String generatedPassword = UUID.randomUUID().toString();
+
+    try {
+      xmppUserService.createUser(username, generatedPassword);
+    } catch (XmppException e) {
+      throw new RegistrationFailedException(e);
```

Comment:
Let the handler of the thrown exception log the exception because the log will be spammed